examples/elasticsearch_helper_example/src/Plugin/ElasticsearchIndex/VersionedIndex.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\elasticsearch_helper_example\Plugin\ElasticsearchIndex;
+
+use Drupal\elasticsearch_helper\Plugin\ElasticsearchIndexBase;
+
+/**
+ * Example versioned index.
+ *
+ * @ElasticsearchIndex(
+ *   id = "versioned_example_index",
+ *   label = @Translation("Example Versioned Index"),
+ *   indexName = "versioned_example{version}",
+ *   typeName = "node",
+ *   entityType = "node",
+ *   versioned = TRUE
+ * )
+ */
+class VersionedIndex extends ElasticsearchIndexBase {
```

Comment:
Should this be abstract class so that its not in plugin cache?

Reply:
This is actually just an example index plugin and would not be loaded unless defined in a module.
Ideally projects that need versioned indexes needs to create similar versioned index plugins.